### PR TITLE
Add and retrieve comments from database

### PR DESCRIPTION
**Pull request by yicyao**
Add comments entered by user in database instead of arraylist so the comments always appear and display in website in bulleted list

Looks like:
db

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -26,30 +32,38 @@
...
 public class DataServlet extends HttpServlet {
-  private List<String> comments = new ArrayList<>();
-
-  // Retrieves previously inputted comments
+  // Retrieves previously inputted comments from database
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    PreparedQuery results = datastore.prepare(new Query("Comment"));
```
Comment (yicyao) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:40]: when I make a new obj (eg new Query) should I inline or make a variable to hold?

Comment (hlz03) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:40]: If the query is not needed elsewhere, we usually inline. 
But I think there are some exceptions, e.g. if a long statement like you need something 
datastore.prepare(QueryBuilder.new().setCommends(new Comments()).setXXX).
for long statement case, I usually make a variable for better readability.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -26,30 +32,38 @@
...
-    response.getWriter().println(json);
     response.sendRedirect("/index.html");
   }
 
   //@return the request parameter, or the default value if the parameter was not specified by the
   // client
   private String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
-    return (value == null) ? defaultValue : value;
+    return (value == null || value.equals("")) ? defaultValue : value;
```
Comment (hlz03) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:67]: value.isEmpty()

Comment (yicyao) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:67]: done

**Pull request merged**